components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java

```diff
@@ -356,8 +364,67 @@ public void updateRoleName(int tenantId, String oldRoleName, String newRoleName)
             }
         } catch (SQLException e) {
             log.error("Error when executing the SQL : " + SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ATT_AND_ATT_VALUE);
-            throw new IdentitySCIMException("Error when reading the SCIM Group information from the " +
-                    "persistence store.", e);
+            throw new IdentitySCIMException("Error when reading the SCIM Group information from the persistence store.",
+                    e);
+        }
+        return roleList.toArray(new String[roleList.size()]);
+    }
+
+    /**
+     * List the groups created from SCIM with a attribute filter and search regex.
+     *
+     * @param searchAttributeName  Search attribute name.
+     * @param searchAttributeValue Search attribute value.
+     * @param tenantId             Tenant ID.
+     * @param domainName           Domain that needs to be filtered.
+     * @return List of SCIM groups.
+     * @throws IdentitySCIMException IdentitySCIMException when reading the SCIM Group information.
+     */
+    public String[] getGroupNameList(String searchAttributeName, String searchAttributeValue, Integer tenantId,
+            String domainName) throws IdentitySCIMException {
+
+        List<String> roleList = new ArrayList<>();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+            try (PreparedStatement prepStmt = connection
+                    .prepareStatement(SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ATT_AND_ATT_VALUE_AND_ROLE_NAME)) {
+                prepStmt.setInt(1, tenantId);
+                prepStmt.setString(2, searchAttributeName);
+                prepStmt.setString(3, searchAttributeValue);
+
+                // Append SQL_FILTERING_DELIMITER to ROLE_NAME param according to the value of domain param.
+                String SQL_FILTERING_DELIMITER = "%";
+                if (StringUtils.isEmpty(domainName)) {
+                    prepStmt.setString(4, SQL_FILTERING_DELIMITER);
```

Comment:
When domain name is empty the isn't the previous query we had is effective

Reply:
Fixed: 788feb46892fd4c42062f6197f51d104401db991